sycl/test-e2e/DeviceLib/string_test.cpp

```diff
@@ -52,6 +52,202 @@ bool kernel_test_memcpy(sycl::queue &deviceQueue) {
+      });
+    });
+  }
+
+  size_t idx;
+  for (idx = 0; idx < 19; ++idx) {
+    if (dst[0][idx] != src[idx])
+      return false;
+  }
+
+  if (dst[0][19] != 'A')
+    return false;
+
+  for (idx = 0; idx < 20; ++idx) {
+    if (dst[1][idx] != src[idx])
+      return false;
+  }
+
+  for (idx = 0; idx < 7; ++idx) {
+    if (dst[2][idx] != src[idx])
+      return false;
+  }
+  for (idx = 7; idx < 11; ++idx) {
+    if (dst[2][idx] != '\0') {
+      return false;
+    }
+  }
+
+  if (dst[2][11] != 'A')
+    return false;
+
+  return true;
+}
+
+class KernelTestStrcmp;
+bool kernel_test_strcmp(sycl::queue &deviceQueue) {
+  int res[5];
+  {
+    sycl::buffer<int, 1> res_buffer(res, sycl::range<1>(5));
+    deviceQueue.submit([&](sycl::handler &cgh) {
+      auto res_acc = res_buffer.get_access<sycl::access::mode::write>(cgh);
+      cgh.single_task<class KernelTestStrcmp>([=]() {
+        char str1[20] = "abcdefg012";
+        char str2[20] = "abcd";
+        char str3[20] = "124ddf";
+        char str4[20] = "abcdefg015";
+        res_acc[0] = strcmp(str1, str1);
+        res_acc[1] = strcmp(str1, str2);
+        res_acc[2] = strcmp(str3, str1);
+        res_acc[3] = strcmp(str4, str1);
+        res_acc[4] = strcmp(str1, str4);
+      });
+    });
+  }
+
+  if ((res[0] != 0) || (res[1] <= 0) || (res[2] >= 0) || (res[3] <= 0) ||
+      (res[4] >= 0))
+    return false;
+  return true;
+}
+
+class KernelTestStrncmp;
+bool kernel_test_strncmp(sycl::queue &deviceQueue) {
+  int res[10];
+  {
+    sycl::buffer<int, 1> res_buffer(res, sycl::range<1>(10));
+    deviceQueue.submit([&](sycl::handler &cgh) {
+      auto res_acc = res_buffer.get_access<sycl::access::mode::write>(cgh);
+      cgh.single_task<class KernelTestStrncmp>([=]() {
+        char str1[20] = "abcdefg012";
+        char str2[20] = "abcd";
+        char str3[20] = "124ddf";
+        char str4[20] = "abcdefg015";
+        char str5[20] = "";
+        char str6[20] = "abcdEFG";
+        char str7[20] = "abcdefg";
+        str6[3] = str7[3] = '\0';
+        res_acc[0] = strncmp(str1, str1, 12);
+        res_acc[1] = strncmp(str1, str2, 4);
+        res_acc[2] = strncmp(str3, str1, 0);
+        res_acc[3] = strncmp(str4, str1, 9);
+        res_acc[4] = strncmp(str1, str4, 10);
+        res_acc[5] = strncmp(str3, str3, 20);
+        res_acc[6] = strncmp(str2, str3, 6);
+        res_acc[7] = strncmp(str5, str5, 12);
+        res_acc[8] = strncmp(str6, str7, 7);
+        str2[0] = str3[0] = '\0';
+        res_acc[9] = strncmp(str2, str3, 4);
+      });
+    });
+  }
+
+  if ((res[0] != 0) || (res[1] != 0) || (res[2] != 0) || (res[3] != 0) ||
+      (res[4] >= 0) || (res[5] != 0) || (res[6] <= 0) || (res[7] != 0) ||
+      (res[8] != 0) || (res[9] != 0))
+    return false;
+  return true;
+}
+
+class KernelTestStrlen;
+bool kernel_test_strlen(sycl::queue &deviceQueue) {
+  bool success = true;
+  char src[20] = "abcdefg012345xyzvvv";
+  size_t len[5] = {
+      0,
+  };
```

Comment:
```suggestion
  size_t len[5] = { 0 };
```

Can we go for this one instead? The comma looks a bit confusing to me.

Reply:
Done.
Thanks very much.